test/epoch.janet

```diff
@@ -0,0 +1,51 @@
+(import testament :prefix "" :exit true)
+
+(import epoch)
+(import test/dtgb)
+
+(deftest leap-year?
+  (is (= (epoch/leap-year? 2000) true))
+  (is (= (epoch/leap-year? 2008) true))
+  (is (= (epoch/leap-year? 1900) false))
+  (is (= (epoch/leap-year? 2019) false))
+  (is (= (epoch/leap-year? 2020) true)))
+
+(deftest timestamp->date
+  (is
+    (=
+     (epoch/timestamp->date 1606717930)
+     {:year 2020
+      :month 10
+      :month-day 29
+      :hours 6
+      :minutes 32
+      :seconds 10
+      :dst false
+      :year-day 334
+      :week-day 1})))
+
+(deftest zero-date
+  (is
+    (=
+     (epoch/timestamp->date 0)
+     {:year 1970
+      :month 0
+      :month-day 0
+      :hours 0
+      :minutes 0
+      :seconds 0
+      :dst false
+      :year-day 0
+      :week-day 4})))
+
+(deftest stdlib-struct
+  (dtgb/assert-is-stdlib-struct! (epoch/timestamp->date 0)))
+
+# This would be cool I think, but somehow it's off by 1 day?
+# Not sure if my implementation is wrong or something's weird with unit testing current time
```

Comment:
```
Expect: {:year-day 336 :minutes 43 :month 11 :dst false :seconds 9 :year 2020 :month-day 0 :hours 18 :week-day 3}
Actual: {:year-day 336 :minutes 43 :month 11 :dst false :seconds 9 :year 2020 :month-day 1 :hours 18 :week-day 3}
````

Today is Dec 2, so `:month-day 1` is correct... That said, it's only `month-day`, not `year-day`.

Reply:
I think it might be as simple as the fact that the reference algorithm, after the second while loop, has a final step:

```
        timep->tm_mday = dayno + 1;
```

I think you just need to `(inc  day_dayno_left)` in the final struct.

Reply:
I replaced this test with one using your `datestr->epoch` function, and it seems to work (I compared using https://www.epochconverter.com/). Maybe something weird about timezones w/ `os/time` vs `os/date`?

I originally took out the `(inc..)` because it would make all the other dates be wrong, e.g. timestamp 0 was day 1, not day 0.

Reply:
@subsetpark I've added a reverse function `date->timestamp`. That one's a little easier. I added tests using your function that shells out to `date`.

Any ideas on epoch vs. unix timestamp vs. timestamp? I kind of mix them up but we should probably pick one.

Reply:
I think of the three, I prefer `epoch` - while it originates from unix, it doesn't NEED to be related to UNIX; and "timestamp" has lots and lots of different meanings. epoch time is the closest thing to a name for the universal system of reckoning time from that particular day in seconds. what do you think?

Reply:
Yea "epoch" seems less specific than UNIX. Just "epoch" or "epoch timestamp?" e.g. for variable names.